nextroute/schema/fleet.go

```diff
@@ -0,0 +1,278 @@
+	Backlog                 []string   `json:"backlog,omitempty"`
+	AlternateStops          []string   `json:"alternate_stops,omitempty"`
+	InitializationCost      int        `json:"initialization_cost,omitempty"`
+}
+
+// FleetStop holds the fleet input stop data.
+// DEPRECATION NOTICE: this part of the API is deprecated and is no longer
+// maintained. It will be deleted soon. Please use [Stop] instead.
+type FleetStop struct {
+	ID                      string       `json:"id,omitempty"`
+	Position                Location     `json:"position,omitempty"`
+	UnassignedPenalty       *int         `json:"unassigned_penalty,omitempty"`
+	Quantity                any          `json:"quantity,omitempty"`
+	Precedes                any          `json:"precedes,omitempty"`
+	Succeeds                any          `json:"succeeds,omitempty"`
+	HardWindow              *[]time.Time `json:"hard_window,omitempty"`
+	MaxWait                 *int         `json:"max_wait,omitempty"`
+	StopDuration            *int         `json:"stop_duration,omitempty"`
+	TargetTime              *time.Time   `json:"target_time,omitempty"`
+	EarlinessPenalty        *float64     `json:"earliness_penalty,omitempty"`
+	LatenessPenalty         *float64     `json:"lateness_penalty,omitempty"`
+	CompatibilityAttributes *[]string    `json:"compatibility_attributes,omitempty"`
+}
+
+// Options adds solver options to the input.
+// DEPRECATION NOTICE: this part of the API is deprecated and is no longer
+// maintained. It will be deleted soon. Please use [solve.Options] instead.
+type Options struct {
+	Solver *SolverOptions `json:"solver,omitempty"`
+}
+
+// SolverOptions represent the solver runtime duration in legacy fleet.
+// DEPRECATION NOTICE: this part of the API is deprecated and is no longer
+// maintained. It will be deleted soon. Please use [solve.Options] instead.
+type SolverOptions struct {
+	Limits *Limits `json:"limits,omitempty"`
+}
+
+// Limits represent the solver runtime limitation in fleet.
+// DEPRECATION NOTICE: this part of the API is deprecated and is no longer
+// maintained. It will be deleted soon. Please use [solve.Options] instead.
+type Limits struct {
+	Duration *CloudDuration `json:"duration,omitempty"`
+}
+
+// CloudDuration is a custom type to represent duration.
+// DEPRECATION NOTICE: this part of the API is deprecated and is no longer
+// maintained. It will be deleted soon. Please use [solve.Options] instead.
+type CloudDuration struct {
+	time.Duration
+}
+
+// ToNextRoute converters a legacy cloud fleet input into nextroute input format.
+func (fleetInput FleetInput) ToNextRoute() (Input, error) {
+	input := Input{
+		Defaults: &Defaults{},
+	}
+	stopCompats := make([]string, 0)
+	vehicleCompats := make([]string, 0)
+	// Use default values and add special handling for CompatibilityAttributes
+	// and HardWindows.
+	if fleetInput.Defaults != nil && fleetInput.Defaults.Stops != nil {
+		input.Defaults = &Defaults{}
```

Comment:
This line can go away because you are doing this now:

```go
	input := Input{
 		Defaults: &Defaults{},
 	}
```

Reply:
done